include/SFML/Audio/Sound.hpp

```diff
@@ -228,10 +230,19 @@ class SFML_AUDIO_API Sound : public SoundSource
     ////////////////////////////////////////////////////////////
     void reattachBuffer();
 
+    ////////////////////////////////////////////////////////////
+    /// \brief Get the sound object
+    ///
+    /// \return The sound object
+    ///
+    ////////////////////////////////////////////////////////////
+    void* getSound() const override;
+
     ////////////////////////////////////////////////////////////
     // Member data
     ////////////////////////////////////////////////////////////
-    const SoundBuffer* m_buffer{}; //!< Sound buffer bound to the source
+    struct Impl;
+    const std::unique_ptr<Impl> m_impl; //!< Implementation details
```

Comment:
Why hide the implementation behind this layer indirection rather than put all of `Sound::Impl`'s data members inside the class directly?

The one reason I can think of doing that is to keep miniaudio headers out of of Sound.hpp.

Reply:
> The one reason I can think of doing that is to keep miniaudio headers out of of Sound.hpp.

Yup, and I think that's a good decision, even if it introduces one extra layer of indirection.